services/src/pro/users/postgres_userdb.rs

```diff
@@ -509,6 +509,33 @@ where
         Ok(())
     }
 
+    async fn bulk_increment_quota_used(&self, quota_used: &[(UserId, u64)]) -> Result<()> {
+        ensure!(self.session.is_admin(), error::PermissionDenied);
+
+        let conn = self.conn_pool.get().await?;
+
+        // collect the user ids and quotas into separate vectors to pass them as parameters to the query
+        let mut users: Vec<UserId> = vec![];
+        let mut quotas: Vec<i64> = vec![];
+        for (user, quota) in quota_used {
+            users.push(*user);
+            quotas.push(*quota as i64);
+        }
+
+        let query = "
+            UPDATE users
+            SET quota_available = quota_available - quota_changes.quota, 
+                quota_used = quota_used + quota_changes.quota
+            FROM 
+                (SELECT * FROM UNNEST($1::uuid[], $2::bigint[]) AS t(id, quota)) AS quota_changes
```

Comment:
ouldn't you insert a two-dim array instead of pushing two arrays.

Reply:
I tried passing it as a Vec of tuples, but that doesn't work. We'd have to implement ToSql for the type `(UserId, i64)`